### PR TITLE
Collection of performance tweaks

### PR DESCRIPTION
**Pull request by petermcs**
Most of them are R6K specific and involve use of new instructions/addressing modes but some are more generally applicible.

### CODE REVIEW

File: Lib/Rabbit4000/libc/string.c [hunk 1]
```diff
@@ -1391,6 +1511,37 @@ __nodebug
 size_t strlen(const char far *s)
 {
 	#asm
+#if _RAB6K
+	align odd                       ; Nicely aligns majority of code on even boundary
```
Comment (tomlogic) [inline Lib/Rabbit4000/libc/string.c]: Can you include similar comments on the other `align` statements?  "Majority of code aligns to even boundary for reduced cycles on 16-bit memory.".

File: Lib/Rabbit4000/RabbitSys/mem.lib [hunk 2]
```diff
@@ -54,21 +54,21 @@ mem_debug
...
+    ld bcde, 1
    ld b, l
    dec b ;; don't include checksum in calc
    xor a
    ld	c,a
-	ld px, (sp+@sp+p)
+   clr hl
 .loop:
-   ld a, (px)
+   ld a, (px+hl)
```
Comment (tomlogic) [inline Lib/Rabbit4000/RabbitSys/mem.lib:65]: I think there should be a comment here explaining the `+hl`.  I assume it's to maintain even alignment for faster execution, but it isn't obvious that it's equivalent to the previous code.

File: Lib/Rabbit4000/libc/string.c [hunk 3]
```diff
@@ -184,18 +238,32 @@ __nodebug
...
+
+	ld		px, pz					; PX = dst (restore the return value)
+#else
+	ld		pz, px					; PZ = PX = dst (saved for return value)
+	ld		py, (sp+@sp+src)		; PY = src
+    clr     hl
+	.cpy:
+	ld		a, (py+hl)				; *dst = *src
+	ld		(px+hl), a				;  (strcpy includes nul terminator)
 	ld		py, py+1					; dst++
```
Comment (tomlogic) [inline Lib/Rabbit4000/libc/string.c:251]: A minor nit, but you could potentially move common code outside of the `#if _RAB6K` block, so it's clear that both platforms have the same entry and exit code, and it's just that non-6000 needs to spell out the `pldi` operation.  Doing so would make it easier to visually compare and confirm the code.

File: Lib/Rabbit4000/libc/string.c [hunk 4]
```diff
@@ -1391,6 +1511,37 @@ __nodebug
 size_t strlen(const char far *s)
 {
 	#asm
+#if _RAB6K
+	align odd                       ; Nicely aligns majority of code on even boundary
+    clr hl
+	ld		py, px					; BCDE = PX = s
```
Comment (tomlogic) [inline Lib/Rabbit4000/libc/string.c]: Shouldn't comment be "PY = PX = s"?

Comment (petermcs) [reply Lib/Rabbit4000/libc/string.c]: Fixed

**Comment by tomlogic** [comment]
Merged in f4e4434.

**Pull request closed without merging**